docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1342,10 +1346,45 @@ The `Digest` message defines the following fields:
   notification using a `P4DataTypeSpec` message (see section on [Representation
   of Arbitrary P4 Types](#sec-representation-of-arbitrary-p4-types)).
 
+### `Extern` { #sec-p4info-extern}
+
+`Extern` messages are used to specify all extern instances across all extern
+types for a non-PSA architecture. This is useful when extending P4Runtime to
+support a new architetcure. Each architecture-specific extern type corresponds
+to at most one `Extern` message instance in P4Info. The `Extern` message defines
+the following fields:
+
+* `extern_type_id`, a 32-bit unsigned integer which uniquely identifies the
+  extern type in the context of the architecture. It must be in the [reserved
+  range](#sec-id-allocation) `[0x81, 0xfe]`. Note that this value does not need
```

Comment:
Is the entire 32-bit value restricted to the range [0x81, 0xfe]?  Or merely the most significant 8 bits or something like that?  Seems like very narrow restriction for a 32-bit value, unless it is 32-bits just for convenience and only values in range [0, 255] are expected to be supported.

Reply:
Except for this small probably-easily-resolved question, which is likely to be more about my ignorance of the wider context than something wrong with the changes, this LGTM.  I am probably the least steeped in the details of P4Runtime of the 3 people you asked to review this, though.

Reply:
This is the type id so yes it is restricted to this range since it is used as the most significant 8 bits of the instance id for each extern instance of this type. It is 32 bits just for convenience (the value actually comes from an enum member).